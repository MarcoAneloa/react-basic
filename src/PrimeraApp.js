import React from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ({ saludo, subtitulo }) => {

    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo,null,2)}</pre> */}
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'soy subtitulo'
}

export default PrimeraApp;