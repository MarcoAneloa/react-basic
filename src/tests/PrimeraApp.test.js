import React from 'react';
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('Prueba en <PrimeraApp />', () => {

    /* test('Debe  mostrar el mensaje "Hola,soy Goku"', () => {
        const saludo = 'Hola, soy Goku';
        const { getByText } = render(<PrimeraApp saludo={saludo} />);
        expect(getByText(saludo)).toBeInTheDocument();
    }); */

    test('Debe  mostrar  <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola,Soy Goku';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe('Soy un subtitulo');
    });


});