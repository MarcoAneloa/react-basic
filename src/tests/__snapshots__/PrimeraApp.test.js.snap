// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp /> Debe  mostrar  <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
     !!
  </h1>
  <p>
    soy subtitulo
  </p>
</Fragment>
`;
